### PR TITLE
add show config statement

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3308
- Other reference link(s):

### CODE REVIEW

File: sql-statements/sql-statement-show-config.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+---
```
Comment (yikeke) [inline sql-statements/sql-statement-show-config.md:1]: Please add this file to TOC.md @jackysp

**Comment by yikeke** [comment]
Please help review this PR. @ran-huang

File: sql-statements/sql-statement-show-config.md (hunk 1 above)
Comment (jackysp) [reply sql-statements/sql-statement-show-config.md:1]: done

File: sql-statements/sql-statement-show-config.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+title: SHOW CONFIG
+summary: Overview of the use of SHOW CONFIG in the TiDB database
+category: reference
+---
+
+# SHOW CONFIG
+
+> **Warning:**
+>
+> This feature is currently an experimental feature and is not recommended for use in a production environment.
```
Comment (ran-huang) [inline sql-statements/sql-statement-show-config.md:11]:
```suggestion
> This feature is currently an experimental feature. It is not recommended to use this feature in the production environment.
```

File: sql-statements/sql-statement-show-config.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+The `SHOW CONFIG` statement is used to show the current configuration of various components of TiDB. Please note that the configuration and system variables act on different dimensions. Please do not confuse them. If you want to obtain system variable information, please use [SHOW VARIABLES](/sql-statements/sql-statement-show-variables.md) syntax.
```
Comment (ran-huang) [inline sql-statements/sql-statement-show-config.md:13]: Please avoid using **please** in docs as much as possible.
```suggestion
The `SHOW CONFIG` statement is used to show the current configuration of various components of TiDB. Note that the configuration and system variables act on different dimensions and should not be mixed up. If you want to obtain the system variable information, use the [SHOW VARIABLES](/sql-statements/sql-statement-show-variables.md) syntax.
```

File: sql-statements/sql-statement-show-config.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
++------+----------------+-------------------------------------------------+---------------------------------------------------------------------+
+| tidb | 127.0.0.1:4000 | advertise-address                               | 127.0.0.1                                                           |
+| tidb | 127.0.0.1:4000 | alter-primary-key                               | false                                                               |
+| tidb | 127.0.0.1:4000 | binlog.binlog-socket                            |                                                                     |
+| tidb | 127.0.0.1:4000 | binlog.enable                                   | false                                                               |
+...
+120 rows in set (0.01 sec)
+```
+
+Show the `type` of the configuration of `tidb`:
```
Comment (ran-huang) [inline sql-statements/sql-statement-show-config.md:47]:
```suggestion
Show the configuration where the `type` is `tidb`:
```

File: sql-statements/sql-statement-show-config.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+```
++------+----------------+-------------------+-----------+
+| Type | Instance       | Name              | Value     |
++------+----------------+-------------------+-----------+
+| tidb | 127.0.0.1:4000 | advertise-address | 127.0.0.1 |
++------+----------------+-------------------+-----------+
+1 row in set (0.05 sec)
+```
+
+You can also use the `LIKE` clause to show the `type` of the configuration of `tidb`:
```
Comment (ran-huang) [inline sql-statements/sql-statement-show-config.md:64]:
```suggestion
You can also use the `LIKE` clause to show the configuration where the `type` is `tidb`:
```

File: sql-statements/sql-statement-show-config.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+| tidb | 127.0.0.1:4000 | alter-primary-key                               | false                                                               |
+| tidb | 127.0.0.1:4000 | binlog.binlog-socket                            |                                                                     |
+| tidb | 127.0.0.1:4000 | binlog.enable                                   | false                                                               |
+...
+40 rows in set (0.01 sec)
+```
+
+## MySQL compatibility
+
+`SHOW CONFIG` is the extended syntax of TiDB, MySQL has no corresponding syntax.
```
Comment (ran-huang) [inline sql-statements/sql-statement-show-config.md:86]:
```suggestion
`SHOW CONFIG` is the extended syntax of TiDB, with no counterpart in MySQL.
```

**Comment by jackysp** [comment]
PTAL @ran-huang

**Comment by ran-huang** [comment]
@bb7133 @qw4990 PTAL~

**Pull request merged**